static/js/controllers/contacts.js

```diff
@@ -151,6 +142,44 @@ var _ = require('underscore'),
             _query();
           }
         }
+      };
+
+      $scope.setupSearchFreetext = function() {
+        SearchFilters.freetext($scope.search);
+      };
+      $scope.setupSearchFacility = function() {
+        SearchFilters.facility(function(facilities) {
+          $scope.filters.facilities = facilities;
+          $scope.search();
+        });
+      };
+      $scope.setupSearchContactType = function() {
+        SearchFilters.contactType(function(types) {
+          $scope.filters.types = types;
+          $scope.search();
+        });
+      };
+      $scope.resetFilterModel = function() {
+        $scope.filters = {};
+        SearchFilters.reset();
+        $scope.search();
+      };
+
+      $scope.search();
+
+      $scope.$on('export', function() {
+        if ($scope.currentTab === 'contacts') {
```

Comment:
That's me being defensive. I'm pretty sure it'll get unloaded so won't be invoked, but there could be a race condition.